### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by Uter1007**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Awesome!!!! there is a conflict though.

File: src/core/client.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import { Rate } from '../models/rate';
+
+export interface IBTCPayClient {
+  get_rates(currencyPairs, storeID): Promise<Rate[]>;
+  create_invoice(payload: any, token?: any): Promise<Invoice>;
+  get_invoice(invoice_id: string, token?: any): Promise<Invoice[]>;
+  get_invoices(params: any, token?: any): Promise<Invoice[]>;
+}
+
+export class BTCPayClient implements IBTCPayClient {
```
Comment (NicolasDorier) [inline src/core/client.ts]: I am not fan of creating an interface which has no concrete use.

Comment (Uter1007) [reply src/core/client.ts]: The thought behind that was to use it for tests & mocking. But its ok for me if we kill it. I am using inversify as dependecy injector in one of my projects and there I am quite often use this kind of strategy.

Comment (Uter1007) [reply src/core/client.ts]: > Awesome!!!! there is a conflict though.

I don't have write access to see the conflict and resolve it.

Comment (Uter1007) [reply src/core/client.ts]: but client.js should not be existent anymore...

Comment (NicolasDorier) [reply src/core/client.ts]: Yeah, I prefer you remove the interface.
1. This is a pain to mock as everytimes a new method is added, the mock need to be udpated
2. This is a pain to maintain as we need to update interface and concrete class
3. This is not really needed, afaik typescript support some sort of duck typing so if your project really need it, you can do it easily and only for the methods you really care.

Comment (Uter1007) [reply src/core/client.ts]: done removed it.

**Comment by NicolasDorier** [comment]
Awesome, @takinbo ?

**Comment by takinbo** [comment]
lgtm! I was able to transpile to js without issues.

**Comment by NicolasDorier** [comment]
Can you fix conflict @Uter1007 https://help.github.com/en/articles/resolving-a-merge-conflict-using-the-command-line

**Comment by Uter1007** [comment]
done

**Comment by NicolasDorier** [comment]
@takinbo seems good to me.
We can later add better typing.

**Comment by Uter1007** [comment]
do you need additional stuff for merging ? cheers, Christoph

**Comment by NicolasDorier** [comment]
@takinbo  can you review / test / make new npm package?

File: package.json [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "tsc"
   },
-  "author": "Tim Akinbo <tim@tanjalo.com>",
+  "author": "Tim Akinbo <tim@tanjalo.com>; Christoph Ott <christoph.ott@lean-coders.at>",
   "license": "MIT",
   "dependencies": {
+    "@types/bs58": "^4.0.0",
```
Comment (NicolasDorier) [inline package.json]: I thought types dependencies were dev time dependencies (Node noob here, am I wrong?)

Comment (Uter1007) [reply package.json]: we could move it - didn't see it as necessary. If you like you can change it. https://stackoverflow.com/questions/45176661/how-do-i-decide-whether-types-goes-into-dependencies-or-devdependencies

**Comment by takinbo** [comment]
Merged. Thanks for the contribution @Uter1007

**Pull request merged**